packages/bigquery/src/index.js

```diff
@@ -76,6 +77,275 @@ function BigQuery(options) {
+    var time;
+
+    if (value.hours) {
+      time = BigQuery.time(value).value;
+    }
+
+    value = format('{y}-{m}-{d}{time}', {
+      y: value.year,
+      m: value.month,
+      d: value.day,
+      time: time ? ' ' + time : ''
+    });
+  }
+
+  this.value = value;
+};
+
+/**
+ * A `TIME` data type represents a time, independent of a specific date.
+ *
+ * @param {object|string} value - The time. If a string, this should be in the
+ *     format the API describes: `[H]H:[M]M:[S]S[.DDDDDD]`. Otherwise, provide
+ *     an object.
+ * @param {string=|number=} value.hours - One or two digits (`00` - `23`).
+ * @param {string=|number=} value.minutes - One or two digits (`00` - `59`).
+ * @param {string=|number=} value.seconds - One or two digits (`00` - `59`).
+ * @param {string=|number=} value.fractional - Up to six digits for microsecond
+ *     precision.
+ *
+ * @example
+ * var time = bigquery.time('14:00:00'); // 2:00 PM
+ *
+ * //-
+ * // Alternatively, provide an object.
+ * //-
+ * var time = bigquery.time({
+ *   hours: 14,
+ *   minutes: 0,
+ *   seconds: 0
+ * });
+ */
+BigQuery.time =
+BigQuery.prototype.time = function(value) {
+  if (!(this instanceof BigQuery.time)) {
+    return new BigQuery.time(value);
+  }
+
+  if (is.object(value)) {
+    value = format('{h}:{m}:{s}{f}', {
+      h: value.hours,
+      m: value.minutes || 0,
+      s: value.seconds || 0,
+      f: is.defined(value.fractional) ? '.' + value.fractional : ''
+    });
+  }
+
+  this.value = value;
+};
+
+/**
+ * A timestamp represents an absolute point in time, independent of any time
+ * zone or convention such as Daylight Savings Time.
+ *
+ * @param {date} value - The time.
+ *
+ * @example
+ * var timestamp = bigquery.timestamp(new Date());
+ */
+BigQuery.timestamp =
+BigQuery.prototype.timestamp = function(value) {
+  if (!(this instanceof BigQuery.timestamp)) {
+    return new BigQuery.timestamp(value);
+  }
+
+  value = value || new Date();
+
+  if (is.date(value)) {
+    value = value.toJSON().replace(/^(.*)T(.*)Z$/, '$1 $2');
+  }
+
+  this.value = value;
+};
+
+/**
+ * Detect a value's type.
+ *
+ * @private
+ *
+ * @resource [Data Type]{@link https://cloud.google.com/bigquery/data-types}
+ *
+ * @param {*} value - The value.
+ * @return {string} - The type detected from the value.
+ */
+BigQuery.getType_ = function(value) {
+  if (value instanceof BigQuery.date) {
+    return 'DATE';
+  }
+
+  if (value instanceof BigQuery.datetime) {
+    return 'DATETIME';
+  }
+
+  if (value instanceof BigQuery.time) {
+    return 'TIME';
+  }
+
+  if (value instanceof BigQuery.timestamp) {
+    return 'TIMESTAMP';
+  }
+
+  if (value instanceof Buffer) {
+    return 'BYTES';
+  }
+
+  if (is.array(value)) {
+    return 'ARRAY';
+  }
+
+  if (is.bool(value)) {
+    return 'BOOL';
+  }
+
+  if (is.number(value)) {
+    return value % 1 === 0 ? 'INT64' : 'FLOAT64';
+  }
+
+  if (is.object(value)) {
+    return 'STRUCT';
+  }
+
+  return 'STRING';
```

Comment:
Might be overkill, but maybe we should do a `is.string()` check here and end with throwing an error for unknown types.